earthnet_models_pytorch/task/loss.py

```diff
@@ -99,56 +96,52 @@ def forward(self, preds, batch, aux, current_step = None):
 class BaseLoss(nn.Module):
     def __init__(self, setting: dict):
         super().__init__()
-
         self.distance = LOSSES[setting["name"]](**setting["args"])
-        self.lambda_state = WeightShedule(**setting["state_shedule"])
-        self.lambda_infer = WeightShedule(**setting["inference_shedule"])
-        self.lambda_l2_res =  WeightShedule(**setting["residuals_shedule"])
+        #self.lambda_state = WeightShedule(**setting["state_shedule"]) # speed of the learning rate ? (I think)
+        #self.lambda_infer = WeightShedule(**setting["inference_shedule"])
+        #self.lambda_l2_res =  WeightShedule(**setting["residuals_shedule"])
         self.dist_scale = 1 if "dist_scale" not in setting else setting["dist_scale"]
         self.ndvi = False if "ndvi" not in setting else setting["ndvi"]
         self.min_lc = 82 if "min_lc" not in setting else setting["min_lc"]
         self.max_lc = 104 if "max_lc" not in setting else setting["max_lc"]
         self.comp_ndvi = True if "comp_ndvi" not in setting else setting["comp_ndvi"]
 
-    def forward(self, preds, batch, aux, current_step = None):
-        
+    def forward(self, preds, batch, aux, current_step = None):   
         logs = {}
-        
-        targs = batch["dynamic"][0][:,-preds.shape[1]:,...]
+        targs = batch["dynamic"][0][:,-preds.shape[1]:,...] 
         if len(batch["dynamic_mask"]) > 0:
             masks = batch["dynamic_mask"][0][:,-preds.shape[1]:,...]
         else:
             masks = None
-        
         if self.ndvi:
-            if targs.shape[2] >= 3 and self.comp_ndvi:
```

Comment:
your editing in this file is not backwards compatible, but i guess it is fine since you want to redo the whole loss API anyway

Reply:
yes, it is unclear, I actually didn't know what exactly you want to encode by self.ndvi, I think it didn't work on en22 with this if, I remove it instead to add a condition because self.comp_ndvi is unclear neither, and I tried to only run a model . What do you think if we change the condition on the shape on a condition to the dataset? And compute the target in the dataloader file avoid having calculations in several places and having dimensions that vary according to the target?